plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java

```diff
@@ -38,21 +48,27 @@ public String getDescription() {
     try {
       CloudSdk cloudSdk = new CloudSdk.Builder().build();
       if (cloudSdk != null) {
-        java.nio.file.Path jarFile = cloudSdk.getJavaAppEngineSdkPath().resolve(API_JAR);
-        if (jarFile != null) {
-          //@formatter:off
-          IClasspathAttribute javadocAttribute = 
-              JavaCore.newClasspathAttribute(IClasspathAttribute.JAVADOC_LOCATION_ATTRIBUTE_NAME, API_JAVADOC_URL);
-          IClasspathEntry appEngineApisEntry = JavaCore.newLibraryEntry(
-              new Path(jarFile.toString()),
-              null /* sourceAttachmentPath */,
-              null /* sourceAttachmentRootPath */,
-              null /* accessRules */,
-              new IClasspathAttribute[] { javadocAttribute },
-              true /* isExported */);
-          //@formatter:on
-          return new IClasspathEntry[] {appEngineApisEntry};
+        List<IClasspathEntry> entries = new ArrayList<>(INCLUDED_JARS.length);
+        for (String jarLocation : INCLUDED_JARS) {
+          java.nio.file.Path jarFile = cloudSdk.getJavaAppEngineSdkPath().resolve(jarLocation);
+          if (jarFile != null) {
+            IClasspathAttribute javadocAttribute = JavaCore.newClasspathAttribute(
+                IClasspathAttribute.JAVADOC_LOCATION_ATTRIBUTE_NAME, APPENGINE_API_JAVADOC_URL);
+            IAccessRule allAccessible =
+                JavaCore.newAccessRule(new Path("**"), IAccessRule.K_ACCESSIBLE);
+            //@formatter:off
+            IClasspathEntry jarEntry = JavaCore.newLibraryEntry(
+                new Path(jarFile.toString()),
+                null /* sourceAttachmentPath */,
+                null /* sourceAttachmentRootPath */,
+                new IAccessRule[] { allAccessible  } /* accessRules */,
+                new IClasspathAttribute[] { javadocAttribute },
+                false /* isExported */);
+            //@formatter:on
+            entries.add(jarEntry);
+          }
         }
+        return entries.toArray(new IClasspathEntry[entries.size()]);
```

Comment:
Looking at the javadoc, there is no practical difference from `entries.toArray()`, isn't it?